lab-06-chris/server.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const Client = require('./model/client');
+const net = require('net');
+const EE = require('events').EventEmitter;
+const ee = new EE();
+const server = module.exports = net.createServer();
+const PORT = process.env.PORT || 3000;
+const pool = [];
+
+ee.on('default', (client, string) => {
+  client.socket.write(`Not a valid command: ${string.split(' ', 1)}\n`);
+});
+
+ee.on('@ALL', (client, string) => {
+  pool.forEach(c => c.socket.write(`Mass message from ${client.nickName}:\n${string}\n`));
+});
+
+ee.on('@NICK', (client, string) => {
+  let oldName = client.nickName;
+  let userEntry = string.split(' ').shift().trim();
+
+  client.nickName = userEntry;
+  pool.forEach(c => c.socket.write(`User: ${oldName}\nChanged name to: ${client.nickName}\n`));
+});
+
+ee.on('@DM', (client, string) => {
+  let fromUser = client.nickName;
+  let msg = string.split(' ').slice(1).join(' ').trim();
+  let holdString = string.split(' ');
+  let userEntry = holdString.shift().trim();
+
+  function findClient(client) {
+    return client.nickName === userEntry;
+  }
+
+  let clientObj = pool.find(findClient);
```

Comment:
This was a really cool use of .find.  When you guys find quirks and doc methods and implement them = win.